src/modules/rust.rs

```diff
@@ -127,24 +127,37 @@ fn find_rust_toolchain_file(context: &Context) -> Option<String> {
     // Look for 'rust-toolchain' as rustup does.
     // https://github.com/rust-lang/rustup.rs/blob/d84e6e50126bccd84649e42482fc35a11d019401/src/config.rs#L320-L358
 
-    fn read_first_line(path: &Path) -> Option<String> {
+    fn read_target(path: &Path) -> Option<String> {
         let content = fs::read_to_string(path).ok()?;
-        let line = content.lines().next()?;
-        Some(line.trim().to_owned())
+        let line = content.lines().next()?.trim();
+
+        if line.is_empty() || line == "[toolchain]" {
+            let value = toml::from_str::<toml::Value>(&content).ok()?;
+            let target = value
+                .as_table()?
```

Comment:
I think it might be better to define two `struct`s here that `#[derive(Deserialize)]`, but this is fine too.

Reply:
I wanted to keep it very simple, but in general I like to have extra structs with derives deserializers for anything involving serde as well. Just wasn't sure what is preferred in starship.

I'll add separate structs for this :+1:

Reply:
I just checked and saw that starship is not using `serde`, so adding it might impact compilation time.

What would you prefer @davidkna?
- Use structs with `#[derive(Deserialize)]` but have to add `serde` as dependency.
- Keep the current version.

Reply:
`serde_derive` is a dependency. Isn't that crate enough to `#[derive(Deserialize)]`? Anyway `toml` already uses `serde` internally and starship also depends on `serde_json` so I don't see any issues with adding `serde` to the dependency list.

If you do need to add a direct dependecy to `serde` maybe remove the `serde_derive` dependency in exchange because it looks like `serde` has superseeded `serde_derive`.

Reply:
You're right, I checked with `cargo tree -i -p serde` and there are several other crates that already include `serde` as dependency.

I just updated the PR using the exact same structs as rustup.